### ISSUE TITLE
Check if input events changes in upcoming Firefox (65) does not break anything

### ISSUE THREAD

**Issue by f1ames** [ckeditor/ckeditor5#1451]
The upcoming Firefox 65 had introduced two changes which can potentially affect how things in editor works (especially typing). We need to check if nothing breaks and everything works smooth in FF 65.

Changes:
* https://www.fxsitecompat.com/en-CA/docs/2018/non-printable-keys-no-longer-fire-keypress-event/
* https://www.fxsitecompat.com/en-CA/docs/2018/keydown-and-keyup-events-are-now-fired-during-ime-composition/

**Comment by f1ames** [comment]
## Events flow

First thing I have checked differences in events flow in Firefox 64, 65 and in Chrome (71) with https://w3c.github.io/uievents/tools/key-event-viewer-ce.html (as Firefox does not fire `beforeinput` event it was filtered out):

**Sequence**: <kbd>f</kbd>,<kbd>Shift</kbd>+<kbd>o</kbd>,<kbd>Backspace</kbd>

Firefox 64
![image](https://user-images.githubusercontent.com/1061942/50892282-6514e580-13fe-11e9-933c-4f85237c1144.png)

Firefox 65
![image](https://user-images.githubusercontent.com/1061942/50892191-3991fb00-13fe-11e9-859b-e607eab3b74c.png)

Chrome 71
![image](https://user-images.githubusercontent.com/1061942/50893905-4a447000-1402-11e9-8d2f-c0735ea4c371.png)

The upcoming Firefox 65 does not fire `keypress` events for non-printable keys as mentioned in the link above. It is the same behaviour as in Chrome so it should not affect editor in any way.

**Sequence (composition - spanish)**: <kbd>´</kbd>,<kbd>a</kbd>,<kbd>Left Arrow</kbd>,<kbd>Delete</kbd>

Firefox 64
![ff64 screen shot 2019-01-09 at 11 11 25](https://user-images.githubusercontent.com/1061942/50892826-abb70f80-13ff-11e9-9c4b-87d90a8d034f.png)

Firefox 65
![ff65 screen shot 2019-01-09 at 11 12 42](https://user-images.githubusercontent.com/1061942/50892834-b1acf080-13ff-11e9-9458-c6cf888bceca.png)

Chrome 71
![image](https://user-images.githubusercontent.com/1061942/50894016-8081ef80-1402-11e9-8b63-577a25abe9b3.png)

Again, the events flow in Firefox 65 got aligned with what Chrome does (apart from the second `input` event which is still fired after `compositionend` in FF) so it should not affect editor, but needs to be checked.

**Sequence (composition - japanese/hiragana)**: <kbd>e</kbd>, <kbd>d</kbd>, <kbd>i</kbd>, <kbd>Arrow Down</kbd>, <kbd>Enter</kbd>, <kbd>Backspace</kbd>, <kbd>Enter</kbd>

Firefox 64
![image](https://user-images.githubusercontent.com/1061942/50896327-04d77100-1409-11e9-9f78-8269b41938d5.png)

Firefox 65
![image](https://user-images.githubusercontent.com/1061942/50896357-1b7dc800-1409-11e9-9308-527c209ce402.png)

Chrome 71
![image](https://user-images.githubusercontent.com/1061942/50896196-a01c1680-1408-11e9-8572-434b2922aa15.png)

Again, Firefox behaviour got aligned with Chrome, so nothing should get broken. One thing to check if we have any Firefox specific "hacks" so we may think about removing them over time.

## The `keypress` event

I haven't found any `keypress` listeners in the code (`keypress` in only mentioned in some [two step caret api docs](https://github.com/ckeditor/ckeditor5-engine/blob/master/src/utils/bindtwostepcarettoattribute.js), but the code uses `keydown` itself) so the removal of `keypress` event for non-printable keys should not be an issue.

When it comes to `keydown` and `keyup` events order, there are few listeners used, especially in code related to typing, but I haven't found anything suspicious which may break. I will also check those cases manually.

## Testing

All unit tests are green in FF 65 on latest `master`. Also when doing manual tests I haven't found any issues.

**To sum up, it seems we are on the safe side with upcoming FF 65. And as the events flow got aligned to Chrome behaviour it's even better for us 😀**
